client/network/src/discovery.rs

```diff
@@ -261,15 +254,32 @@ impl DiscoveryBehaviour {
 
 	/// Call this method when a node reports an address for itself.
 	///
-	/// **Note**: It is important that you call this method, otherwise the discovery mechanism will
-	/// not properly work.
-	pub fn add_self_reported_address(&mut self, peer_id: &PeerId, addr: Multiaddr) {
+	/// **Note**: It is important that you call this method. The discovery mechanism will not
+	/// automatically add connecting peers to the Kademlia k-buckets.
+	pub fn add_self_reported_address(&mut self, peer_id: &PeerId, protocols: &[String], addr: Multiaddr) {
```

Comment:
This is a publicly-exposed method, so `protocols` should be documented and should have a more generic API in my opinion.

```suggestion
	///
	/// `supported_protocols` is a list of protocol names that the peer supports.
	pub fn add_self_reported_address(&mut self, peer_id: &PeerId, supported_protocols: impl Iterator<Item = impl AsRef<str>>, addr: Multiaddr) {
```